mailbox/noise.go

```diff
@@ -685,7 +713,10 @@ func (b *Machine) GenActTwo() ([ActTwoSize]byte, error) {
 
 	authPayload := b.EncryptAndHash(payload[:])
 
-	actTwo[0] = HandshakeVersion
+	// The responder will always send its preferred handshake version in
+	// the hopes that the initiator will then upgrade to this version in
```

Comment:
Upgrade to the version in act 2? I think this assumes a lot w.r.t how new versions will be rolled out all together.

Reply:
The reasoning behind this flow is that in our case, the initiator (TW) will always be updated before the responder (Litd) . And since the initiator is also the one that sends the first message (act 1), it makes this negotiation a bit tricky. So we went with this approach: 
- Act 1: client sends its lowest acceptable version in act 1 (client cant send its max version here since then every new handshake version would be a breaking change and everyone would need to upgrade their Lit nodes)
- Act 2: server sends its max version. Client will get this and will know which version the server will be happy with and will adapt to that version. 
- Act 3: client sends back what ever the server sent in act 2.

Let me know if there is a better way 👍